### PR TITLE
[IOS-379]: Improve Yap inconsistency handling

### PR DESCRIPTION
**Pull request by yuliaveres**
After second try just clean whatever is left there, password or database, and proceed to fresh login flow.

### CODE REVIEW

File: Toshi/Models/Yap.swift [hunk 1]
```diff
@@ -57,10 +74,22 @@ public final class Yap: NSObject, Singleton {
         return UserDefaults.standard.bool(forKey: UserDB.password)
     }
 
+    @objc public static var inconsistentStateDescription: String? {
+        return inconsistencyError.description
+    }
```
Comment (roberthein) [inline Toshi/Models/Yap.swift]: You can also write it shorter like this:
```
@objc public static var inconsistentStateDescription = inconsistencyError.description
```

File: Toshi/Models/Yap.swift [hunk 2]
```diff
@@ -57,10 +74,22 @@ public final class Yap: NSObject, Singleton {
...
+
+    @objc public static var inconsistencyError: YapInconsistencyError {
+        if isUserDatabaseFileAccessible {
+            return .missingKeychainPassword
+        }
+
+        return .missingDatabaseFile
+    }
```
Comment (roberthein) [inline Toshi/Models/Yap.swift]: You can also write this shorter like this:
```
@objc public static var inconsistencyError: YapInconsistencyError = isUserDatabaseFileAccessible ? .missingKeychainPassword : .missingDatabaseFile
```

File: Toshi/Models/Yap.swift [hunk 3]
```diff
@@ -92,20 +121,28 @@ public final class Yap: NSObject, Singleton {
...
+        removeDatabaseFileAndPassword()
+    }
+
+    fileprivate func removeDatabaseFileAndPassword() {
+        KeychainSwift().delete(UserDB.password)
+        UserDefaults.standard.removeObject(forKey: UserDB.password)
+
+        self.deleteFileIfNeeded(at: UserDB.dbFilePath)
+        self.deleteFileIfNeeded(at: UserDB.walFilePath)
+        self.deleteFileIfNeeded(at: UserDB.shmFilePath)
```
Comment (roberthein) [inline Toshi/Models/Yap.swift]: `self` `self` `self`

File: Toshi/Models/Yap.swift [hunk 4]
```diff
@@ -20,6 +20,23 @@ protocol Singleton: class {
...
+    case missingDatabaseFile
+    case missingKeychainPassword
+
+    var description: String? {
+        switch self {
+        case .missingDatabaseFile:
+            return Localized("yap_missing_db_file_error_description")
+        case .missingKeychainPassword:
+            return Localized("yap_missing_password_error_description")
+        default:
```
[the remark this answers is not in the document]
Comment (yuliaveres) [reply Toshi/Models/Yap.swift]: sure

**Pull request merged**